trlx/trainer/accelerate_base_trainer.py

```diff
@@ -353,9 +353,9 @@ def evaluate(self):  # noqa: C901
 
             stats["time/generate"] = time() - generate_time
 
-            samples = self.accelerator.gather(torch.vstack(all_samples))
-            prompts = self.accelerator.gather(torch.vstack(all_prompts))
-            prompt_sizes = self.accelerator.gather(torch.hstack(prompt_sizes))
+            samples = self.accelerator.gather_for_metrics(torch.vstack(all_samples))
+            prompts = self.accelerator.gather_for_metrics(torch.vstack(all_prompts))
+            prompt_sizes = self.accelerator.gather_for_metrics(torch.hstack(prompt_sizes))
```

Comment:
This constraint would be benign anyway since nothing else depends on accelerate, however we have to check whether there isn't something broken with this new release to be deliberate with the version here, so probably let's add it later